### ISSUE TITLE
Not clear how to prepare custom data for training

### ISSUE THREAD

**Issue by JNaranjo-Alcazar** [Megvii-BaseDetection/YOLOX#818]
Hello community!

I have a question regarding the preparation of data to train this network with my own data. Right now I have two folders (images and labels) with the same name per file. The labels are in xml (VOC). I don't understand very well the training manual and I don't know how to continue. What exactly do I have to change in the `exps/example/custom/yolox_s.py`?

A question I have at this point is, what exactly are these files?

```
self.train_ann = "instances_train2017.json".
self.val_ann = "instances_val2017.json"
```

Best regards

**Comment by sunmooncode** [comment]
If the label is xml(VOC), you have two ways to train the model .
1:you can make the data to VOC format, then use **exps/yolox_voc/yolox_voc_s.py** and change the data address.
2:you can aslo change the data to coco format,  there is a tool to help you complete it.  **https://github.com/RapidAI/YOLO2COCO**

**Comment by ImSuMyatNoe** [comment]
@JNaranjo-Alcazar  I also do not know how to train the custom dataset. Have you already found out? If so, Could you please tell me how to do it? :(

**Comment by hongsamvo** [comment]
In case your dataset in VOC format, you can follow the author's guide, remember to change your dataset name [https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/exps/example/yolox_voc/yolox_voc_s.py#L46](train) and [https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/exps/example/yolox_voc/yolox_voc_s.py#L104](val), in voc.py [https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/yolox/data/datasets/voc.py#L105](here),  and make sure no error at [https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/yolox/data/datasets/voc.py#L124](here),  change [https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/yolox/data/datasets/voc_classes.py#L6](your classes)

**Comment by ImSuMyatNoe** [comment]
Thank you very much for your help @hongsamvo . Does " ln -s /path/to/your/VOCdevkit ./datasets/VOCdevkit"  can be run on the Pycharm linux terminal?

**Comment by hongsamvo** [comment]
I think it can

**Comment by JNaranjo-Alcazar** [comment]
Thanks for the reply @hongsamvo. I tried to follow the guide but is quite confusing. So, just to make it clear, following your instructions, we should be able to train the yolox model?

Thanks again

**Comment by hongsamvo** [comment]
yes, actually the error will happen [here](https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/yolox/data/datasets/voc.py#L125) and other similar places, just replace the term `"VOC" + year` with your dataset name, and one error [here](https://github.com/Megvii-BaseDetection/YOLOX/blob/38c633bf176462ee42b110c70e4ffe17b5753208/yolox/data/datasets/voc.py#L325), choose metric you want.
